nextjs/src/app/reports/[id]/useBoundaryAnalytics.ts

```diff
@@ -37,12 +41,16 @@ const useBoundaryAnalytics = (
     dataOutputKey = 'importedDataCountByConstituency'
   } else if (boundaryType === 'admin_ward') {
     query = MAP_REPORT_WARD_STATS
+    // @ts-ignore — asserting here that importedDataCountByWard will also return GroupedDataCount[]
     dataOutputKey = 'importedDataCountByWard'
   } else throw new Error('Invalid boundary type')
 
-  const boundaryAnalytics = useQuery(query, { variables, skip: !canQuery })
+  const boundaryAnalytics = useQuery<MapReportConstituencyStatsQuery>(query, {
```

Comment:
I think it's too confusing and messy to leave it like this - to use a type belonging to a specific query (or at least named so).

Reply:
Check out the new changes, I'm quite happy with em :)